### DISCUSSION TITLE
Container with freely positioned children?

### DISCUSSION THREAD

**Discussion by johannesvollmer** [hecrj/iced#969] [Ideas]
Hi! I'm trying to build a [node editor](https://devtalk.blender.org/t/pbr-node-tree/12876). The user should be able to place and drag some nodes around on in infinite container. Later, the container should also support scaling and translating all children at once, in order to navigate the node graph. 

I couldn't find a Container suitable for this, as all containers would automatically layout their children. Is there a container for this? Is there a guide on Iced yet, where I could lookup something like this? I would love to contribute such a widget if there is none yet.

**Comment by manokara** [comment]
There's no free positioning container indeed, but the usual approach to a node editor would be with the `Canvas` widget. The [Game of Life example](https://github.com/hecrj/iced/tree/master/examples/game_of_life) might be a good starting point. However, since you can't use widgets inside of a Canvas, you'll have to create all controls from scratch.

**Reply by johannesvollmer** [reply]
Thanks for your the hints. Using the Canvas was indeed an option I considered. However, as the nodes are rather complex, I hope it will be easier to code a custom layout container than to re-implement all the widgets inside the nodes using the Canvas.
